libs/core/iterator_support/include/hpx/iterator_support/counting_iterator.hpp

```diff
@@ -183,4 +186,19 @@ namespace hpx { namespace util {
     {
         return counting_iterator<Incrementable>(x);
     }
-}}    // namespace hpx::util
+
+    namespace detail {
+        template <typename Incrementable>
+        using counting_shape_type = hpx::util::iterator_range<
+            hpx::util::counting_iterator<Incrementable>>;
+
+        template <typename Incrementable>
+        HPX_HOST_DEVICE inline counting_shape_type<Incrementable>
+        make_counting_shape(Incrementable n)
+        {
+            return hpx::util::make_iterator_range(
+                hpx::util::make_counting_iterator(Incrementable(0)),
+                hpx::util::make_counting_iterator(n));
+        }
+    }    // namespace detail
+}}       // namespace hpx::util
```

Comment:
#5429 moves this facility to it's own file, see https://github.com/STEllAR-GROUP/hpx/pull/5429/files#diff-c3c03e6827ab3b06f782481b097c19238540e0a89ef723a46982430a2944a7ef.

Reply:
We're on the same page then :) Do you mind the move to `detail`? I'd go with your PR first. I can rebase this on yours.

Reply:
Having it in namespace detail is fine.